### DISCUSSION TITLE
startNotifications

### DISCUSSION THREAD

**Discussion by dgnovo** [capacitor-community/bluetooth-le#498] [General]
Hello, 
i want to use with a Ionic Capacitor app, the startNotifications function to receive notifications on a Custom Service.

I receive this message:
**Setting notification failed**
when i use
`
await BleClient.startNotifications(
      device.deviceId,
      service,
     characteristic,
      (value) => {
        console.log('current value', value);
      }
    );
`

The Custom Service doesn't have a descriptor on **charateristc** and **notify** property is set to **true**.

I see on Device.kt this error is at the line
`
val descriptor = characteristic.getDescriptor(UUID.fromString(CLIENT_CHARACTERISTIC_CONFIG))
        if (descriptor == null) {
            reject(key, "Setting notification failed.")
            return
        }
`

Is it possible use startNotification if charateristc doesn't have a descriptor?

**Comment by pwespi** [comment]
> Is it possible use startNotification if charateristc doesn't have a descriptor?

No, this is currently not possible.

What peripheral is this? Do you control the software/firmware of the peripheral?
